spec/presenters/tree_builder_miq_action_cat_spec.rb

```diff
@@ -0,0 +1,72 @@
+describe TreeBuilderMiqActionCat do
+  before do
+    role = MiqUserRole.find_by_name("EvmRole-operator")
+    @group = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Tags Group")
+    login_as FactoryGirl.create(:user, :userid => 'tags_wilma', :miq_groups => [@group])
+    @tag1 = FactoryGirl.create(:classification, :name => 'tag1', :show => false)
+    @folder1 = FactoryGirl.create(:classification, :name => 'folder1', :show => true)
+    @folder1.entries.push(@tag1)
+    @tag2 = FactoryGirl.create(:classification, :name => 'tag2', :show => false)
+    @folder2 = FactoryGirl.create(:classification, :name => 'folder2', :show => true)
+    @folder2.entries.push(@tag2)
+    @group = FactoryGirl.create(:miq_group)
+    @tenant = "TestTenant"
+    @tenant = "#{@tenant} Tags"
+  end
+  context 'read-only mode' do
```

Comment:
newlines please ;)

And you *probably* want to look into using `let` and `let!`, instead of instance variables here..